Integrated-Windows-Auth-Flow/src/main/java/IntegratedWindowsAuthFlow.java

```diff
@@ -1,82 +1,74 @@
-
-       return  future.get();
+        IAuthenticationResult result = acquireTokenIwa();
+        System.out.println("Access token: " + result.accessToken());
+        System.out.println("Id token: " + result.idToken());
+        System.out.println("Account username: " + result.account().username());
     }
 
-    private static String getUsersListFromGraph(String accessToken) throws IOException {
-        URL url = new URL("https://graph.microsoft.com/v1.0/users");
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+    private static IAuthenticationResult acquireTokenIwa() throws Exception {
 
-        conn.setRequestMethod("GET");
-        conn.setRequestProperty("Authorization", "Bearer " + accessToken);
-        conn.setRequestProperty("Accept", "application/json");
+        // Load token cache from file and initialize token cache aspect. The token cache will have
+        // dummy data, so the acquireTokenSilently call will fail.
+        TokenCacheAspect tokenCacheAspect = new TokenCacheAspect("sample_cache.json");
 
-        int httpResponseCode = conn.getResponseCode();
-        if(httpResponseCode == 200) {
+        PublicClientApplication pca = PublicClientApplication.builder(CLIENT_ID)
+                .authority(AUTHORITY)
+                .setTokenCacheAccessAspect(tokenCacheAspect)
+                .build();
 
-            StringBuilder response;
-            try(BufferedReader in = new BufferedReader(
-                    new InputStreamReader(conn.getInputStream()))){
+        Set<IAccount> accountsInCache = pca.getAccounts().join();
+        // Take first account in the cache. In a production application, you would filter
+        // accountsInCache to get the right account for the user authenticating.
+        IAccount account = accountsInCache.iterator().next();
 
-                String inputLine;
-                response = new StringBuilder();
-                while (( inputLine = in.readLine()) != null) {
-                    response.append(inputLine);
-                }
+        IAuthenticationResult result;
+        try {
+            SilentParameters silentParameters =
+                    SilentParameters
+                            .builder(SCOPE, account)
+                            .build();
+
+            // try to acquire token silently. This call will fail since the token cache
+            // does not have any data for the user you are trying to acquire a token for
+            result = pca.acquireTokenSilently(silentParameters).join();
+        } catch (Exception ex) {
+            if (ex.getCause() instanceof MsalException) {
+
+                IntegratedWindowsAuthenticationParameters parameters =
+                        IntegratedWindowsAuthenticationParameters
+                                .builder(SCOPE, USER_NAME)
+                                .build();
+
+                // Try to acquire a IWA. You will need to generate a Kerberos ticket.
```

Comment:
it is acquiring tokens using IWA

Reply:
Adjusted that comment to fix that typo in the latest commit. However, it was copied from a sample in the main library repository, so I'll change it there as well in the near future.